accord-core/src/test/java/accord/impl/list/ListExecutor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package accord.impl.list;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import accord.local.CommandStore;
+import accord.local.PreLoadContext;
+import accord.utils.async.AsyncChain;
+
+public interface ListExecutor
```

Comment:
not a fan of this, wondering if we should make a `AsyncExecutor` and have `CommandStore` implement as well?